types/react-relay/index.d.ts

```diff
@@ -11,20 +11,34 @@ export {
     commitLocalUpdate,
     commitRelayModernMutation as commitMutation,
     fetchRelayModernQuery as fetchQuery,
+    GraphQLTaggedNode,
     requestRelaySubscription as requestSubscription,
 } from "relay-runtime";
 
 import * as React from "react";
 import * as RelayRuntimeTypes from "relay-runtime";
 
+// ~~~~~~~~~~~~~~~~~~~~~
+// Utility types
+// ~~~~~~~~~~~~~~~~~~~~~
+
+// Taken from https://github.com/pelotom/type-zoo
+// tslint:disable-next-line:strict-export-declare-modifiers
+type Diff<T extends string, U extends string> = ({ [P in T]: P } & { [P in U]: never } & { [x: string]: never })[T];
+// tslint:disable-next-line:strict-export-declare-modifiers
+type Omit<T, K extends keyof T> = Pick<T, Diff<keyof T, K>>;
+
+export type RemoveRelayProp<P> = Omit<P & { relay: never }, "relay">;
+
+export type ComponentRef = { componentRef?: (ref: any) => void };
+
+export type RelayContainer<T> = React.ComponentType<RemoveRelayProp<T> & ComponentRef>;
```

Comment:
Decided to refactor all typings related to containers into this one location after all.